sourmash/minhash.py

```diff
@@ -0,0 +1,675 @@
+                            to_bytes(codon)).decode('utf-8')
+        except SourmashError as e:
+            raise ValueError(e.message)
+
+    def count_common(self, other, downsample=False):
+        """\
+        Return the number of hashes in common between ``self`` and ``other``.
+
+        Optionally downsample ``scaled`` objects to highest ``scaled`` value.
+        """
+        if not isinstance(other, MinHash):
+            raise TypeError("Must be a MinHash!")
+        return self._methodcall(lib.kmerminhash_count_common, other._get_objptr(), downsample)
+
+    def downsample(self, num=None, scaled=None):
+        """Copy this object and downsample new object to either `num` or
+        `scaled`.
+        """
+        if num is None and scaled is None:
+            raise ValueError('must specify either num or scaled to downsample')
+        elif num is not None:
+            if self.num and self.num < num:
+                raise ValueError("new sample num is higher than current sample num")
+            max_hash=0
+        elif scaled is not None:
+            if self.num:
+                raise ValueError("num != 0 - cannot downsample a standard MinHash")
+            max_hash = self.max_hash
+            if max_hash is None:
+                raise ValueError("no max_hash available - cannot downsample")
+
+            old_scaled = _get_scaled_for_max_hash(self.max_hash)
+            if old_scaled > scaled:
+                raise ValueError(
+                    "new scaled {} is lower than current sample scaled {}".format(
+                        scaled, old_scaled
+                    )
+                )
+
+            max_hash = _get_max_hash_for_scaled(scaled)
+            num = 0
+        ###
+
+        # create new object:
+        a = MinHash(
+            num, self.ksize, self.is_protein, self.dayhoff, self.hp,
+            self.track_abundance, self.seed, max_hash
+        )
+        # copy over hashes:
+        if self.track_abundance:
+            a.set_abundances(self.get_mins(with_abundance=True))
+        else:
+            a.add_many(self)
+
+        return a
+
+    @deprecated(deprecated_in="3.5", removed_in="4.0",
+                current_version=VERSION,
+                details='Use downsample(num=...) instead.')
+    def downsample_n(self, new_num):
+        "Copy this object and downsample new object to num=``new_num``."
+        return self.downsample(num=new_num)
+
+    @deprecated(deprecated_in="3.5", removed_in="4.0",
+                current_version=VERSION,
+                details='Use scaled instead.')
+    def downsample_max_hash(self, *others):
+        """Copy this object and downsample new object to min of ``*others``.
+
+        Here, ``*others`` is one or more MinHash objects.
+        """
+        max_hashes = [x.max_hash for x in others]
+        new_max_hash = min(self.max_hash, *max_hashes)
+        new_scaled = get_scaled_for_max_hash(new_max_hash)
```

Comment:
```suggestion
        new_scaled = _get_scaled_for_max_hash(new_max_hash)
```
(this is a bug that [sonarcloud](https://sonarcloud.io/project/issues?id=dib-lab_sourmash&open=AXOQyD9y8hpWe-mDGL3C&pullRequest=1128&resolved=false&types=BUG) caught)